doctr/documents/reader.py

```diff
@@ -17,11 +19,55 @@
 AbstractFile = Union[AbstractPath, bytes]
 Bbox = Tuple[float, float, float, float]
 
+def orient_image(img):
+    gray_img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+    gray_img = cv2.medianBlur(gray_img, 5)
+    thresh = cv2.threshold(gray_img, thresh=0, maxval=255, type=cv2.THRESH_BINARY_INV + cv2.THRESH_OTSU)[1]
+
+    # try to merge words in lines
+    (h, w) = img.shape[:2]
+    k_x = max(1, (floor(w / 100)))
+    k_y = max(1, (floor(h / 100)))
+    kernel = cv2.getStructuringElement(cv2.MORPH_RECT, (k_x, k_y))
+    thresh = cv2.dilate(thresh, kernel, iterations=1)
+
+    # extract contours
+    contours, _ = cv2.findContours(thresh, cv2.RETR_LIST, cv2.CHAIN_APPROX_SIMPLE)
+
+    # Sort contours
+    contours = sorted(contours, key=cv2.contourArea, reverse=True)
+
+    n_ct = 50
+    ratio_threshold_for_lines = 5
+
+    angles = []
+    for contour in contours[:n_ct]:
+        _, (w, h), angle = cv2.minAreaRect(contour)
+        if w / h > ratio_threshold_for_lines:  # select only contours with ratio like lines
+            angles.append(angle)
+        elif w / h < 1 / ratio_threshold_for_lines:  # if lines are vertical, substract 90 degree
+            angle -= 90
+            angles.append(angle)
+    angle = median(angles)
```

Comment:
This part of the function might be better as a standalone function `estimate_orientation` and we'll need to add some typing annotations :sweat_smile:

Reply:
Since this isn't a "reading" feature, I'd suggest either moving it to `doctr.models.utils` or `doctr.documents.utils`

Reply:
OK, then I think it will fit well inside `doctr.models._utils`